pilot/pkg/xds/eds.go

```diff
@@ -170,6 +170,19 @@ func (s *DiscoveryServer) edsCacheUpdate(shard model.ShardKey, hostname string,
 	return fullPush
 }
 
+func (s *DiscoveryServer) RemoveShard(shardKey model.ShardKey) {
```

Comment:
This is not enough, looks at the `deleteService`, also need UpdateServiceAccount

Reply:
re-using `deleteService` now